pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -568,34 +568,57 @@ func (r *ReconcileLogStorage) Reconcile(ctx context.Context, request reconcile.R
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileLogStorage) elasticsearchSecrets(ctx context.Context) ([]*corev1.Secret, error) {
+// Checks if public cert secret managed by ECK is valid. If not valid, the cert is deleted. Returns true if the cert was deleted.
+func (r *ReconcileLogStorage) deleteInvalidECKManagedPublicCertSecret(ctx context.Context, secret *corev1.Secret, certKey string, svcDNSNames []string) (bool, error) {
+	ok, err := render.SecretHasExpectedDNSNames(secret, certKey, svcDNSNames)
+	if err != nil {
+		return false, err
```

Comment:
If I get this right .... If the secret of the cert is missing data (has the keys, but does not have data) we do not need to create it and we return an error ?

Reply:
This cert is created by the ECK operator and we're only deleting it in this case where the cert's DNS names are invalid. Deleting the secret should mean that when the ECK operator recreates it, it will have the updated DNS names.